packages/tools/client-debugger/client-debugger-view/src/components/client-data-views/ContainerHistoryRecord.tsx

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { DefaultPalette, IStackItemStyles, Icon, Stack, StackItem } from "@fluentui/react";
+import React from "react";
+
+import {
+	ConnectionStateChangeLogEntry,
+	ContainerStateChangeKind,
+} from "@fluid-tools/client-debugger";
+
+/**
+ * Input props describing a Fluid {@link IFluidClientDebugger.getContainerConnectionLog}.
+ */
+export interface ContainerHistoryRecordViewProps {
+	/**
+	 * The status history of container to display.
+	 */
+	containerHistory: readonly ConnectionStateChangeLogEntry[];
+}
+
+/**
+ * Displays basic information about the provided {@link ContainerHistoryProps.containerHistory}.
+ */
+export function ContainerHistoryRecordView(
+	props: ContainerHistoryRecordViewProps,
+): React.ReactElement {
+	const { containerHistory } = props;
+	const nowTimeStamp = new Date();
+
+	// Reverse history such that newest events are displayed first
+	const reversedHistoryLog = [...containerHistory].reverse();
+
+	const historyViews: React.ReactElement[] = [];
+
+	for (const changeEntry of reversedHistoryLog) {
+		const changeTimeStamp = new Date(changeEntry.timestamp);
+		const wasChangeToday = nowTimeStamp.getDate() === changeTimeStamp.getDate();
+
+		const accordianBackgroundColor: IStackItemStyles = {
+			root: {
+				background:
+					changeEntry.newState === ContainerStateChangeKind.Connected ||
+					changeEntry.newState === ContainerStateChangeKind.Attached
+						? "#90ee90"
+						: "#FF7377",
+				borderStyle: "solid",
+				borderWidth: 1,
+				borderColor: DefaultPalette.neutralTertiary,
+				padding: 3,
+			},
+		};
+
+		const iconStyle: IStackItemStyles = {
+			root: {
+				padding: 10,
+			},
+		};
+
+		historyViews.push(
+			<div>
+				<Stack horizontal={true} styles={accordianBackgroundColor}>
+					<StackItem styles={iconStyle}>
+						<Icon
+							iconName={
+								changeEntry.newState === ContainerStateChangeKind.Connected
+									? "PlugConnected"
+									: changeEntry.newState === ContainerStateChangeKind.Attached
+									? "Attach"
+									: changeEntry.newState === ContainerStateChangeKind.Disconnected
+									? "PlugDisconnected"
+									: changeEntry.newState === ContainerStateChangeKind.Disposed
+									? "RemoveLink"
+									: "SkypeCircleMinus"
```

Comment:
Let's make the last check explicitly for `Closed`, and then default to an "unknown" kind of icon (maybe a question mark) in case we receive something that we don't expect.

Reply:
Modified